docs/visual/faq.md

```diff
@@ -0,0 +1,143 @@
+
+### Why is Screener not picking up data from my CI build?
+For screener-storybook and screener-runner tests run within a CI tool, Screener will automatically set build, branch, and commit options if you are using one of the following CI tools:
+
+* Jenkins
+* CircleCI
+* Travis CI
+* Visual Studio Team Services
+* Codeship
+* GitLab CI
+* Drone
+* Bitbucket Pipelines
+* Semaphore
+* Buildkite
+
+This works by pulling data from environment variables in the CI job.
+<details><summary><strong>Click here to view sample code containing expected Environment variables</strong></summary>
+
+```js reference title="Screener Runner"
+https://github.com/screener-io/screener-runner/blob/master/src/ci.js
+```
+
+</details>
+
+As an example, Jenkins looks for the following environment variables: `JENKINS_URL` or `JENKINS_HOME`, and `BUILD NUMBER`:
+```js reference title="Screener Runner"
+https://github.com/screener-io/screener-runner/blob/master/src/ci.js#L23
+```
+
+CircleCI looks for the following environment variables: `CI`, `CIRCLECI`, `CIRCLE_BUILD_NUM`:
+```js reference title="Screener Runner"
+https://github.com/screener-io/screener-runner/blob/master/src/ci.js#L31
+```
+
+If these environment variables are NOT accessible in the build, the reason is most likely that the CI is running the test job inside a separate container, and the variables need to be passed into the container.
+
+### How do I add Screener Steps to Storybook Component Story Format (CSF) stories?
+Screener Storybook supports version 3.0 to 6.x (latest) and [CSF](https://storybook.js.org/docs/react/api/csf)-formatted stories. For more information, see [Three ways to add Screener Steps to a React Component in Storybook to CSF](https://gist.github.com/screener-io/dfbd4b9aa5284e555ea83c936fc0b2ba).
+
+### How do I add Screener Steps to my Storybook project that uses React Hooks?
+React Hooks messes up using the Screener component because it executes its magic at runtime. So to use Screener Steps with React Hooks, you need to use Storybook Decorators to wrap the component story. Here is an example script: [Storybook + Screener Steps with Hooks](https://gist.github.com/screener-io/ada0e3c51137c907ba80728ad7fab23b).
+
+### Does Sauce Visual interact with knobs within stories in Storybook and test as needed?
+Screener-storybook does not automatically interact with knobs, as there would be potentially an infinite number of permutations. but if there's a UI state that the user wants to get to using knobs, then this is possible with Screener Steps.
+
+In the Storybook knobs add-on, there's a knobs UI where the user can copy a query string for the knobs configuration. They can copy this querystring, and use it with Screener Steps `executeScript` method to reproduce those knobs values.
+
+Click on the copy button in the lower right, then paste into an editor, then copy only the query-string portion (after the `?`) and paste that into a Screener step.
```

Comment:
This "click on the copy button" is it a functionality provided by the editor?

Reply:
@luishernandezv I believe this is referring to the storybook editor. You can update the knob position in the UI and then copy the resulting querystring like here: https://storybooks-official.netlify.app/?path=/story/addons-knobs-withknobs--tweaks-static-values-organized-in-groups. It's a bit tough to describe without a screenshot, but for users of storybook I think they will understand.